kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/UserSortNameComparator.java

```diff
@@ -56,6 +57,16 @@ public int compare(User u1, User u2) {
         if (u1 == null) return (nullsLow ? -1 : 1);
         if (u2 == null) return (nullsLow ? 1 : -1);
 
-        return new NullSafeComparator<>(collator, nullsLow).compare(u1.getSortName(), u2.getSortName());
+        // Replace spaces to handle sorting scenarios where surname has space
+        String prop1 = StringUtils.replace(u1.getSortName(), " ", "+");
+        String prop2 = StringUtils.replace(u2.getSortName(), " ", "+");
```

Comment:
Does adjusting the Collator make more sense than a separate String manipulation?

Reply:
I find this horribly unreadable:

oldRules.replaceAll("<'\u005f'", "<' '<'\u005f'");

Does it make sense to you? is it not working as expected?

Reply:
yeah you could make the code easier to read